io/src/lib.rs

```diff
@@ -0,0 +1,225 @@
+    /// to a hashed open string and save this move(first character from string) to determine the winners.
+    ///
+    /// # Arguments:
+    /// * `String`: is the raw string of move("0" or "1" or "2" or "3" or "4") + "password" that should be equal to string that was sent in `MakeMove(String)` without hashing.
+    ///
+    /// # Requirements:
+    /// * The hashed(by program) `Reveal` string must be equal to this round `MakeMove` string.
+    /// * The `GameStage` must be `GameStage::Reveal(StageDesciption)` where `StageDescription::anticipated_players` must contains `msg::source()`
+    ///
+    /// On success replies `Event::SuccessfulMove(ActorId)` where `ActorId` is the moved player's address.
+    Reveal(String),
+
+    /// Changes the game config of the next game.
+    /// When the current game ends, this config will be applied.
+    ///
+    /// # Arguments:
+    /// * `GameConfig`: is the config that will be applied to the next game.
+    ///
+    /// # Requirements:
+    /// * The `msg::source()` must be the owner of the program.
+    /// * `players_count_limit` of the `GameConfig` must be greater than 1
+    /// * `entry_timeout` of the `GameConfig` must be greater than 5000(5 sec)
+    /// * `move_timeout` of the `GameConfig` must be greater than 5000(5 sec)
+    /// * `reveal_timeout` of the `GameConfig` must be greater than 5000(5 sec)
+    ///
+    /// On success replies `Event::GameConfigChanged`.
+    ChangeNextGameConfig(GameConfig),
+
+    /// Stops the game.
+    /// This action can be used, for example, to change the configuration of the game,
+    /// or if the players have gone on strike and do not want to continue playing,
+    /// or if the game has gone on for a long time.
+    /// When the admin stops the game, all funds are distributed among the players remaining in the game.
+    /// If the game is in the registration stage, bets will be returned to the entire lobby.
+    ///
+    /// # Requirements:
+    /// * The `msg::source()` must be the owner of the program.
+    ///
+    /// On success replies `Event::GameWasStopped(BTreeSet<ActorId>)` where inside are the players who got the money.
+    StopGame,
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum Event {
+    PlayerRegistred,
+    SuccessfulMove(ActorId),
+    SuccessfulReveal(RevealResult),
+    GameConfigChanged,
+    GameStopped(BTreeSet<ActorId>),
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum State {
+    Config,
+    LobbyList,
+    GameStage,
+    CurrentStageTimestamp,
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum StateReply {
+    Config(GameConfig),
+    LobbyList(Vec<ActorId>),
+    GameStage(GameStage),
+    CurrentStageTimestamp(u64),
+}
+
+#[derive(Debug, Default, Clone, Encode, Decode, TypeInfo, PartialEq)]
+pub struct GameConfig {
+    pub bet_size: u128,
+    pub players_count_limit: u8,
+    pub entry_timeout: u64,  // in ms
+    pub move_timeout: u64,   // in ms
+    pub reveal_timeout: u64, // in ms
```

Comment:
Better to add units to name instead of comments.
```suggestion
    pub entry_timeout_ms: u64,
    pub move_timeout_ms: u64,
    pub reveal_timeout_ms: u64,
```